packages/core/dialect/saphana/index.ts

```diff
@@ -0,0 +1,207 @@
+
+import {
+  ConnectionDialect
+} from '@sqltools/core/interface';
+import * as Utils from '@sqltools/core/utils';
+import queries from './queries';
+import GenericDialect from '@sqltools/core/dialect/generic';
+import { DatabaseInterface } from '@sqltools/core/plugin-api';
+
+interface Statement {
+  exec(params: any[], handler: (err: any, row: any) => void);
+}
+
+interface HanaClientModule {
+  createConnection(connOptions: object): HanaConnection;
+}
+
+interface HanaConnection {
+  connect(handler: (err: any) => void);
+  exec(query: string, handler: (err: any, row: any) => void);
+  disconnect();
+  prepare(query: string, handler: (err: any, statement: Statement) => void);
+}
+
+export default class SAPHana extends GenericDialect<HanaConnection> implements ConnectionDialect {
+
+  public static deps: typeof GenericDialect['deps'] = [{
+    type: 'package',
+    name: '@sap/hana-client',
+    version: '2.4.126'
+  }];
+
+  private get lib() {
+    return __non_webpack_require__('@sap/hana-client') as HanaClientModule;
+  }
+
+  queries = queries;
+  private schema: String;
+
+  public open(encrypt?: boolean): Promise<HanaConnection> {
+    if (this.connection) {
+      return this.connection;
+    }
+
+    this.needToInstallDependencies();
+
+    const connOptions = {
+      HOST: this.credentials.server,
+      PORT: this.credentials.port,
+      UID: this.credentials.username,
+      PWD: this.credentials.password
+    }
+    if (this.credentials.connectionTimeout && this.credentials.connectionTimeout > 0) {
+      connOptions["CONNECTTIMEOUT"] = this.credentials.connectionTimeout * 1000;
+    }
+    if (encrypt) {
+      connOptions["ENCRYPT"] = true;
+    }
+
+    try {
+      let conn = this.lib.createConnection(connOptions);
+
+      this.connection = new Promise<HanaConnection>((resolve, reject) => conn.connect(err => {
+        if (err) {
+          console.error("Connection to HANA failed", err.toString());
+          reject(err);
+        }
+        this.schema = this.credentials.database;
+        conn.exec("SET SCHEMA " + this.schema, err => {
+            if (err) {
+              reject(err);
+            }
+            console.log("Connection to SAP Hana succedded!");
+            resolve(conn);
+          });
+      }));
+      return this.connection;
+    } catch (e) {
+      console.error("Connection to HANA failed" + e.toString());
+      Promise.reject(e);
+    }
+  }
+
+  public async close() {
+    if (!this.connection) return Promise.resolve();
+
+    await this.connection.then(conn => conn.disconnect());
+    this.connection = null;
+  }
+
+  public async testConnection?() {
+    return this.open().then(conn => conn.exec('select 1 from dummy;', (err, rows) => rows));
```

Comment:
is it a real query? I dont know if dummy exists for all SAPHana dbs, I would suggest to use something like `select 1` or `select CURRENT_DATE`

Reply:
yes it is a real query and always there

Reply:
thanks!